src/biogeophys/SnowHydrologyMod.F90

```diff
@@ -480,7 +482,11 @@ subroutine BulkDiag_NewSnowDiagnostics(bounds, num_c, filter_c, &
        int_snow(c) = max(int_snow(c),h2osno_total(c)) !h2osno_total could be larger due to frost
 
        ! snowmelt from previous time step * dtime
-       snowmelt(c) = qflx_snow_drain(c) * dtime
+       ! add h2osfc to ice conversion flux to snowmelt to trigger
+       ! re-calculation of scf when h2osfc freezes (snowmelt
+       ! amount is not used in sfc_method, just presence/absence)
+       snowmelt(c) = (qflx_h2osfc_to_ice(c) + qflx_snow_drain(c)) * dtime
```

Comment:
This was just used as a trigger to get the SCF to be updated.  Normally, it is updated when newsnow > 0 or snowmelt > 0.  If neither occurs, then SCF is not updated.  I'm not sure, but I think the latter behavior was intentional, and due to undesirable behavior caused by frost deposition or surface water freezing.  But now it seems like it would be good to update SCF for changes to due to surface water freezing.

Another way to do this would be to have a second call to scf_method%UpdateSnowDepthAndFrac where the newsnow argument is set to qflx_h2osfc_to_ice and bifall is set to denice.  I think that is conceptually what is happening.  I can test that.